xarray/tests/test_backends.py

```diff
@@ -2213,6 +2212,77 @@ class PyNioTestAutocloseTrue(PyNioTest):
     autoclose = True
 
 
+@requires_netCDF4
+@requires_pnc
+class TestPseudoNetCDF(CFEncodedDataTest, NetCDF3Only, TestCase):
+    def test_write_store(self):
+        # pseudonetcdf is read-only for now
+        pass
+
+    @contextlib.contextmanager
+    def open(self, path, **kwargs):
+        with open_dataset(path, engine='pseudonetcdf',
+                          autoclose=self.autoclose,
+                          **kwargs) as ds:
+            yield ds
+
+    def save(self, dataset, path, **kwargs):
+        dataset.to_netcdf(path, engine='netcdf4', **kwargs)
+
+    def test_weakrefs(self):
+        example = Dataset({'foo': ('x', np.arange(5.0))})
+        expected = example.rename({'foo': 'bar', 'x': 'y'})
+
+        with create_tmp_file() as tmp_file:
+            example.to_netcdf(tmp_file, engine='netcdf4')
+            on_disk = open_dataset(tmp_file, engine='pseudonetcdf')
+            actual = on_disk.rename({'foo': 'bar', 'x': 'y'})
+            del on_disk  # trigger garbage collection
+            assert_identical(actual, expected)
+
+    @pytest.mark.skip(reason='cannot pickle file objects')
+    def test_pickle(self):
+        pass
+
+    @pytest.mark.skip(reason='cannot pickle file objects')
+    def test_pickle_dataarray(self):
+        pass
+
+    @pytest.mark.skip(reason='does not support boolean dtype')
+    def test_roundtrip_boolean_dtype(self):
+        # pnc does not support boolean dtype
+        pass
+
+    @pytest.mark.skip(reason='does not support auto masking and scaling')
+    def test_roundtrip_mask_and_scale(self):
+        # pnc does not support auto masking and scaling
+        pass
+
+    @pytest.mark.skip(reason='does not support object types')
+    def test_roundtrip_object_dtype(self):
+        # pnc does not support object types
```

Comment:
Also done by xarray -- no backends support storing object dtype

Reply:
Fails on my machine

```
    def test_roundtrip_object_dtype(self):
        floats = np.array([0.0, 0.0, 1.0, 2.0, 3.0], dtype=object)
        floats_nans = np.array([np.nan, np.nan, 1.0, 2.0, 3.0], dtype=object)
        bytes_ = np.array([b'ab', b'cdef', b'g'], dtype=object)
        bytes_nans = np.array([b'ab', b'cdef', np.nan], dtype=object)
        strings = np.array([u'ab', u'cdef', u'g'], dtype=object)
        strings_nans = np.array([u'ab', u'cdef', np.nan], dtype=object)
        all_nans = np.array([np.nan, np.nan], dtype=object)
        original = Dataset({'floats': ('a', floats),
                            'floats_nans': ('a', floats_nans),
                            'bytes': ('b', bytes_),
                            'bytes_nans': ('b', bytes_nans),
                            'strings': ('b', strings),
                            'strings_nans': ('b', strings_nans),
                            'all_nans': ('c', all_nans),
                            'nan': ([], np.nan)})
        expected = original.copy(deep=True)
        with self.roundtrip(original) as actual:
            try:
                assert_identical(expected, actual)
            except AssertionError:
                # Most stores use '' for nans in strings, but some don't.
                # First try the ideal case (where the store returns exactly)
                # the original Dataset), then try a more realistic case.
                # This currently includes all netCDF files when encoding is not
                # explicitly set.
                # https://github.com/pydata/xarray/issues/1647
                expected['bytes_nans'][-1] = b''
                expected['strings_nans'][-1] = u''
>               assert_identical(expected, actual)
E               AssertionError: <xarray.Dataset>
E               Dimensions:       (a: 5, b: 3, c: 2)
E               Dimensions without coordinates: a, b, c
E               Data variables:
E                   floats        (a) object 0.0 0.0 1.0 2.0 3.0
E                   floats_nans   (a) object nan nan 1.0 2.0 3.0
E                   bytes         (b) object b'ab' b'cdef' b'g'
E                   bytes_nans    (b) object b'ab' b'cdef' b''
E                   strings       (b) object 'ab' 'cdef' 'g'
E                   strings_nans  (b) object 'ab' 'cdef' 
E                   all_nans      (c) object nan nan
E                   nan           float64 nan
E               <xarray.Dataset>
E               Dimensions:       (a: 5, b: 3, c: 2)
E               Dimensions without coordinates: a, b, c
E               Data variables:
E                   floats        (a) float64 0.0 0.0 1.0 2.0 3.0
E                   floats_nans   (a) float64 nan nan 1.0 2.0 3.0
E                   bytes         (b) |S4 b'ab' b'cdef' b'g'
E                   bytes_nans    (b) |S4 b'ab' b'cdef' b''
E                   strings       (b) object 'ab' 'cdef' 'g'
E                   strings_nans  (b) object 'ab' 'cdef' ''
E                   all_nans      (c) float64 nan nan
E                   nan           float64 0.0

xarray/tests/test_backends.py:325: AssertionError
```

Reply:
Maybe this is not pnc?